test/index.test.ts

```diff
@@ -257,6 +257,71 @@ describe("mkdist", () => {
     `);
   }, 50_000);
 
+  it("mkdist (emit sourcemaps)", async () => {
+    const rootDir = resolve(__dirname, "fixture");
+    const { writtenFiles } = await mkdist({
+      rootDir,
+      esbuild: {
+        sourcemap: "linked",
```

Comment:
I only added a test case with the `"linked"` option since that's what I'm personally interested in, and I wasn't sure how the option would change if `esbuild` were to be swapped out with `rolldown`. I'd be happy to add test cases for the other modes if deemed necessary.